lib/platform/github/gh-graphql-wrapper.ts

```
@@ -0,0 +1,118 @@
+    this.repoItemFilterBy = itemFilterBy;
+  }
+
+  setRepoItemNodeList(itemNodeList: string[]): void {
+    this.repoItemNodeList = itemNodeList;
+  }
+
+  setRepoResultNumEls(numEls: number): void {
+    this.repoResultNumEls = numEls;
+  }
+
+  async get(cursor = ''): Promise<any> {
+    const url = 'graphql';
+    const headers = {
+      accept: 'application/vnd.github.merge-info-preview+json',
+    };
+    const afterCursor = cursor === '' ? null : `"${cursor}"`;
+    // prettier-ignore
+    const nodeListString = this.repoItemNodeList.join('\n');
+    const query = `
+    query {
+      repository(owner: "${this.repoOwner}", name: "${this.repoName}") {
+        ${this.repoItem}(first: ${
+      this.repoResultNumEls
+    }, after:${afterCursor}, orderBy: {field: UPDATED_AT, direction: DESC}, filterBy: ${
+      this.repoItemFilterBy
+    }) {
+          pageInfo {
+            startCursor
+            hasNextPage
+          }
+          nodes {
+            ${nodeListString}
+          }
+        }
+      }
+    }
+    `;
+
+    const options = {
+      headers,
+      body: JSON.stringify({ query }),
+      json: false,
+    };
+
+    try {
+      const res = JSON.parse((await (get as any).post(url, options)).body);
+
+      if (!res.data) {
+        // retry query until numElements == 1
+        if (this.repoResultNumEls === 1) {
+          logger.info({ query, res }, 'No graphql res.data');
+          return [false, [], ''];
+        }
+
+        this.repoResultNumEls = parseInt(
+          (this.repoResultNumEls / 2).toString(),
+          10
+        );
+        return await this.get(cursor);
+      }
+
+      const nextCursor = res.data.repository[this.repoItem].pageInfo.hasNextPage
+        ? res.data.repository[this.repoItem].pageInfo.startCursor
+        : '';
+
+      return [true, res.data.repository[this.repoItem].nodes, nextCursor];
+    } catch (err) {
+      logger.warn({ query, err }, 'graphql.get error');
+      return [false, [], ''];
```

Let's throw a `platform-error` here instead of returning empty.